editcomponents.cgi

```diff
@@ -229,30 +244,42 @@ if ($action eq 'update') {
     'initialcc'        => {'type' => 'multi'},
   });
 
-  my $comp_old_name      = trim($cgi->param('componentold')     || '');
-  my $default_bug_type   = trim($cgi->param('default_bug_type') || '');
-  my $default_assignee   = trim($cgi->param('initialowner')     || '');
-  my $default_qa_contact = trim($cgi->param('initialqacontact') || '');
-  my $description        = trim($cgi->param('description')      || '');
-  my $triage_owner       = trim($cgi->param('triage_owner')     || '');
-  my $team_name          = trim($cgi->param('team_name')        || '');
-  my @initial_cc         = $cgi->param('initialcc');
-  my $isactive           = $cgi->param('isactive');
-  my $bug_desc_template  = $cgi->param('bug_description_template'),
+  my $comp_old_name = trim($cgi->param('componentold') || '');
 
   my $component
     = Bugzilla::Component->check({product => $product, name => $comp_old_name});
 
-  $component->set_name($comp_name);
-  $component->set_description($description);
-  $component->set_default_bug_type($default_bug_type);
-  $component->set_default_assignee($default_assignee);
-  $component->set_default_qa_contact($default_qa_contact);
-  $component->set_triage_owner($triage_owner);
-  $component->set_team_name($team_name);
-  $component->set_cc_list(\@initial_cc);
-  $component->set_is_active($isactive);
-  $component->set_bug_description_template($bug_desc_template);
+  if ($editcomponents) {
+    my $default_bug_type   = trim($cgi->param('default_bug_type') || '');
+    my $default_assignee   = trim($cgi->param('initialowner')     || '');
+    my $default_qa_contact = trim($cgi->param('initialqacontact') || '');
+    my $description        = trim($cgi->param('description')      || '');
+    my $team_name          = trim($cgi->param('team_name')        || '');
+    my @initial_cc         = $cgi->param('initialcc');
+    my $isactive           = $cgi->param('isactive');
+    my $bug_desc_template  = $cgi->param('bug_description_template'),
+
+    $component->set_name($comp_name);
+    $component->set_description($description);
+    $component->set_default_bug_type($default_bug_type);
+    $component->set_default_assignee($default_assignee);
+    $component->set_default_qa_contact($default_qa_contact);
+    $component->set_team_name($team_name);
+    $component->set_cc_list(\@initial_cc);
+    $component->set_is_active($isactive);
+    $component->set_bug_description_template($bug_desc_template);
+  }
+
+  if ($edittriageowners) {
+    # This should really be fixed one day
+    if (!$editcomponents) {
```

Comment:
Here's the `edittriageowners` footgun - right now it's impossible for `edittriageowners` to be true and `editcomponents` to be false.